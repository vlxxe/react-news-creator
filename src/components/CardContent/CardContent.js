import React, { useEffect } from "react"
import { useDispatch, useSelector } from "react-redux"

import { fetchCardContent } from "../../store/actions/cardsActions"
import { EditCard } from "../EditCard/EditCard"

export const CardContent = ({ cardId }) => {
  const dispatch = useDispatch()

  useEffect(() => {
    dispatch(fetchCardContent(cardId))
  }, [cardId, dispatch])

  const cardContent = useSelector(state => state.cards.cardContent)

  if (cardContent === null) {
    return (
      <div>
        <h2>
          Загрузка <div uk-spinner=""></div>
        </h2>
      </div>
    )
  }

  return (
    <div className="uk-card uk-card-default">
      <div className="uk-card-header">
        <div className="uk-grid-small uk-flex-middle" uk-grid="">
          <div className="uk-width-auto">
            <img
              className="uk-border-circle"
              width="40"
              height="40"
              src={cardContent.profile.img}
              alt="user-pic"
            />
          </div>
          <div className="uk-width-expand">
            <h3 className="uk-card-title uk-margin-remove-bottom">
              {cardContent.title}
              <EditCard
                cardAuthorId={cardContent.profile.id}
                cardTitle={cardContent.title}
                cardId={cardId}
              />
            </h3>
            <div className="uk-flex uk-flex-between">
              <p className="uk-text-meta uk-margin-remove-top uk-margin-remove-bottom">
                {cardContent.profile.name}
              </p>
              <span className="uk-text-meta">{cardContent.date}</span>
            </div>
          </div>
        </div>
      </div>
      <div className="uk-card-body">
        <p>{cardContent.text}</p>
      </div>
    </div>
  )
}
