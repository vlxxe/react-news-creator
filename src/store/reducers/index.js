import { combineReducers } from "redux";
import { googleAuthReducer } from "./googleAuthReducer";
import { cardsReducer } from "./cardsReducer";
import { reducer as formReducer } from "redux-form";

export const rootReducer = combineReducers({
  googleAuth: googleAuthReducer,
  cards: cardsReducer,
  form: formReducer
});
