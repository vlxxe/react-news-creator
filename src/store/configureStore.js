import { createStore, applyMiddleware } from "redux"
import { rootReducer } from "./reducers"
import logger from "redux-logger"
import thunk from "react-thunk"

export const store = createStore(rootReducer, applyMiddleware(logger))
