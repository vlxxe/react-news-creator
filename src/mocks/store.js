import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureStore([thunk]);

export let emptyStore = mockStore({
  googleAuth: {
    userProfile: {
      id: null,
      name: null,
      img: null
    }
  },
  cards: {
    cardsList: [],
    cardContent: null,
    isLoading: false
  }
});
