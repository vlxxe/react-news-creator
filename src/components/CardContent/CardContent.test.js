import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";

import { CardContent } from "./CardContent";
import { emptyStore } from "../../mocks/store";
import { shallow, mount } from "enzyme";

describe("CardContent", () => {
  let component = renderer.create(
    <Provider store={emptyStore}>
      <Router>
        <CardContent />
      </Router>
    </Provider>
  );

  it("render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("show loader without data", () => {
    expect(component).toContainEqual(<h2>Загрузкаа</h2>);
  });
});
