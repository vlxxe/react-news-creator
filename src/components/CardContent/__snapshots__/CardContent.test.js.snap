// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContent render correctly 1`] = `
<div>
  <h2>
    Загрузка 
    <div
      uk-spinner=""
    />
  </h2>
</div>
`;
